packages/babel-plugin-transform-react-jsx-self/src/index.js

```diff
@@ -16,11 +16,78 @@ import { types as t } from "@babel/core";
 
 const TRACE_ID = "__self";
 
+/**
+ * Returns whether the given path contains a `super()` call.
+ */
+function containsSuperCall(sourcePath) {
+  // We don't want to go into inner classes as their super() calls aren't relevant for us.
+  if (sourcePath.isClassDeclaration() || sourcePath.isClassExpression()) {
+    return false;
+  }
+  let exists = false;
+  sourcePath.traverse({
+    // Again, we don't go into inner classes.
+    ClassDeclaration(path) {
+      path.skip();
+    },
+    ClassExpression(path) {
+      path.skip();
+    },
+    // Find a `Super` that is used in a call expression.
+    Super(path) {
+      if (path.parentPath.isCallExpression()) {
+        exists = true;
+        path.stop();
+      }
+    },
+  });
+  return exists;
+}
+
+/**
+ * Returns whether it is allowed to use `this` at given path.
+ */
+function canUseThis(path) {
+  const statement = path.getStatementParent();
```

Comment:
Did you mean `path.scope.getFunctionParent().path`?

e.g.

```
class B {}
class A extends B {
  constructor(a) {
    {
      <span>{a}</span>
    }
    super();
  }
}
```
In this case `statement` will be a `BlockStatement` which does not contain a super call.

Reply:
Done.